openedx/features/pakx/lms/overrides/utils.py

```diff
@@ -444,6 +446,27 @@ def get_course_progress_percentage(request, course_key):
     return _calculate_progress_for_block(get_progress_information(request, course_key))
 
 
+def get_and_save_course_progress(course_key, user_id):
+    """Calculate and save a learner's progress"""
+    from .models import CourseProgressStats
+
+    user = User.objects.get(id=user_id)
+    request = RequestFactory().request()
+    request.user = user
+
+    progress_percentage = float(get_course_progress_percentage(request, course_key))
+
+    try:
+        course_progress_stats = CourseProgressStats.objects.get(enrollment__user=user, enrollment__course=course_key)
```

Comment:
simply update the progress
`CourseProgressStats.objects.filter(enrollment__user=user, enrollment__course=course_key).update(progress=progress_percentage)`